apps/gateway/test/champions_test.exs

```diff
@@ -295,6 +298,121 @@ defmodule Gateway.Test.Champions do
     end
   end
 
+  describe "campaigns" do
+    test "get campaigns and levels", %{socket_tester: socket_tester} do
+      # Register user
+      {:ok, user} = Users.register("campaign_user")
+
+      # GetCampaigns
+      SocketTester.get_campaigns(socket_tester, user.id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{
+        response_type: {:campaigns, %Campaigns{} = campaigns}
+      }
+
+      assert Enum.count(campaigns.campaigns) > 0
+
+      sample_campaign = Enum.random(campaigns.campaigns)
+
+      # GetCampaign
+      SocketTester.get_campaign(socket_tester, user.id, sample_campaign.id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{
+        response_type: {:campaign, %Campaign{}}
+      }
+
+      fetch_last_message(socket_tester)
+
+      %WebSocketResponse{
+        response_type: {:campaign, %Campaign{} = campaign_to_verify}
+      } =
+        get_last_message()
+
+      assert campaign_to_verify.id == sample_campaign.id
+
+      # GetLevel
+      level = Enum.random(sample_campaign.levels)
+
+      SocketTester.get_level(socket_tester, user.id, level.id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{
+        response_type: {:level, %Level{}}
+      }
+
+      fetch_last_message(socket_tester)
+
+      %WebSocketResponse{
+        response_type: {:level, %Level{} = level_to_verify}
+      } = get_last_message()
+
+      assert level_to_verify.id == level.id
+    end
+
+    test "fight level", %{socket_tester: socket_tester} do
+      # Register user
+      {:ok, user} = Users.register("battle_user")
+
+      # Get user's first campaign progression
+      [campaign_progression | _] = user.campaign_progress
+
+      # Get the level of the campaign progression
+      level_id = campaign_progression.level_id
+
+      # FightLevel
+      SocketTester.fight_level(socket_tester, user.id, level_id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{
+        response_type: {:battle_result, _}
+      }
+
+      fetch_last_message(socket_tester)
+
+      %WebSocketResponse{
+        response_type: {:battle_result, _ = battle_result}
+      } = get_last_message()
+
+      # Battle result should be either win or loss
+      assert battle_result.result == "win" or battle_result.result == "loss"
+
```

Comment:
We should check that the campaign progression is updated correctly to the next level! when we win

Reply:
Maybe we could do this in another test, where we set up a strong team to ensure victory (since we don´t have seeds yet)

Reply:
Done!